addons/account/models/account_move.py

```diff
@@ -1483,7 +1483,7 @@ def _compute_amount(self):
                 elif currency.compare_amounts(total_to_pay, total_residual) != 0:
                     new_pmt_state = 'partial'
 
-            if new_pmt_state == 'paid' and move.move_type in ('in_invoice', 'out_invoice', 'entry'):
+            if new_pmt_state == 'paid' and move.move_type in ('in_invoice', 'out_invoice', 'entry') and not self._context.get('skip_move_reverse_state'):
```

Comment:
Instead of using a context key in a touchy and very important compute function, wouldn't it be safer to extract the `search` from the loop by  searching for all ids in `self` at the same time and mapping the results?

Reply:
That's an interesting approach, we can try that. Could you help me on implementing this ?  Maybe if you have an example of this somewhere else in the code that would be cool :)

Reply:
Something like
```python
in_invoices = self.filtered(lambda m: m.move_type == 'in_invoice')
out_invoices = self.filtered(lambda m: m.move_type == 'out_invoice')
others = self.filtered(lambda m: m.move_type not in ('in_invoice', 'out_invoice'))
reversed_mapping = defaultdict(self.env['account.move'])
for move in self.env['account.move'].search([
    ('state', '=', 'posted'), 
    '|', '|',
    '&', ('reversed_entry_id', 'in', in_invoices.ids, ('move_type', '=', 'in_refund'),
    '&', ('reversed_entry_id', 'in', out_invoices.ids, ('move_type', '=', 'out_refund'),
    '&', ('reversed_entry_id', 'in', others.ids, ('move_type', '=', 'entry'),
]):
    reversed_mapping[move.reversed_entry_id] += move
```

Then you can access inside of the loop like this
```python
                reverse_moves = reversed_mapping[move]
```

I didn't test any of this, but something similar can be done if `company.tax_exigibility` to find the CABA moves